planning/behavior_velocity_planner/autoware_behavior_velocity_blind_spot_module/src/util.cpp

```diff
@@ -0,0 +1,314 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <autoware/behavior_velocity_blind_spot_module/util.hpp>
+#include <autoware/behavior_velocity_planner_common/utilization/path_utilization.hpp>
+#include <autoware_lanelet2_extension/utility/utilities.hpp>
+
+#include <boost/geometry/algorithms/area.hpp>
+#include <boost/geometry/algorithms/distance.hpp>
+#include <boost/geometry/algorithms/length.hpp>
+
+#include <lanelet2_core/geometry/Polygon.h>
+
+namespace autoware::behavior_velocity_planner
+{
+
+namespace
+{
+static bool hasLaneIds(
+  const tier4_planning_msgs::msg::PathPointWithLaneId & p, const lanelet::Id id)
+{
+  for (const auto & pid : p.lane_ids) {
+    if (pid == id) {
+      return true;
+    }
+  }
+  return false;
+}
+
+static std::optional<std::pair<size_t, size_t>> findLaneIdInterval(
+  const tier4_planning_msgs::msg::PathWithLaneId & p, const lanelet::Id id)
+{
+  bool found = false;
+  size_t start = 0;
+  size_t end = p.points.size() > 0 ? p.points.size() - 1 : 0;
+  if (start == end) {
+    // there is only one point in the path
+    return std::nullopt;
+  }
+  for (size_t i = 0; i < p.points.size(); ++i) {
+    if (hasLaneIds(p.points.at(i), id)) {
+      if (!found) {
+        // found interval for the first time
+        found = true;
+        start = i;
+      }
+    } else if (found) {
+      // prior point was in the interval. interval ended
+      end = i;
+      break;
+    }
+  }
+  start = start > 0 ? start - 1 : 0;  // the idx of last point before the interval
+  return found ? std::make_optional(std::make_pair(start, end)) : std::nullopt;
+}
+}  // namespace
+
+std::optional<InterpolatedPathInfo> generateInterpolatedPathInfo(
+  const lanelet::Id lane_id, const tier4_planning_msgs::msg::PathWithLaneId & input_path,
+  rclcpp::Logger logger)
+{
+  constexpr double ds = 0.2;
+  InterpolatedPathInfo interpolated_path_info;
+  if (!splineInterpolate(input_path, ds, interpolated_path_info.path, logger)) {
+    return std::nullopt;
+  }
+  interpolated_path_info.ds = ds;
+  interpolated_path_info.lane_id = lane_id;
+  interpolated_path_info.lane_id_interval =
+    findLaneIdInterval(interpolated_path_info.path, lane_id);
+  return interpolated_path_info;
+}
+
+std::optional<lanelet::ConstLanelet> getSiblingStraightLanelet(
+  const lanelet::Lanelet assigned_lane,
+  const lanelet::routing::RoutingGraphConstPtr routing_graph_ptr)
+{
+  for (const auto & prev : routing_graph_ptr->previous(assigned_lane)) {
+    for (const auto & following : routing_graph_ptr->following(prev)) {
+      if (std::string(following.attributeOr("turn_direction", "else")) == "straight") {
+        return following;
+      }
+    }
+  }
+  return std::nullopt;
+}
+
+lanelet::ConstLanelet generateHalfLanelet(
+  const lanelet::ConstLanelet lanelet, const TurnDirection & turn_direction,
+  const double ignore_width_from_centerline)
```

Comment:
- [X] same impl as before
- [x] check `planner_param_.ignore_width_from_centerline` is passed as before